cloudslang-content-maven-compiler/src/main/java/io/cloudslang/maven/compiler/CloudSlangMavenCompiler.java

```diff
@@ -48,86 +50,180 @@ public boolean canUpdateTarget(CompilerConfiguration configuration) throws Compi
 
     @Override
     public CompilerResult performCompile(CompilerConfiguration config) throws CompilerException {
+        init(config);
         CompilerResult compilerResult = new CompilerResult();
         List<CompilerMessage> compilerMessage = new ArrayList<>();
 
         //we do not want the source files that were calculated because we have multiple suffix
         //and the framework support only one via the inputFileEnding
         config.setSourceFiles(null);
         String[] sourceFiles = getSourceFiles(config);
-        if (sourceFiles.length > 0) {
-            System.out.println("Compiling " + sourceFiles.length + " " + "source file" + (sourceFiles.length == 1 ? "" : "s"));
+        Map<String,byte[]> dependenciesSourceFiles = getDependenciesSourceFiles(config) ;
+        if(sourceFiles.length > 0) {
+            System.out.println("Compiling " + sourceFiles.length + " " + "source file" + (sourceFiles.length == 1?"":"s"));
             for (String sourceFile : sourceFiles) {
-                compilerMessage.addAll(compileFile(sourceFile));
+                compilerMessage.addAll(compileFile(sourceFile,sourceFiles,dependenciesSourceFiles));
             }
 
-            if (compilerMessage.size() > 0) {
+            if (compilerMessage.size() > 0){
                 compilerResult.setCompilerMessages(compilerMessage);
-                compilerResult.setSuccess(false);
+                //we want to set it to false only in case we want to fail the build
+                if (errorLevel.equals(CompilerMessage.Kind.ERROR)){
+                    compilerResult.setSuccess(false);
+                }
             }
         }
 
         return compilerResult;
     }
 
-    private List<CompilerMessage> compileFile(String sourceFile) {
-        ExecutableModellingResult executableModellingResult;
-        List<CompilerMessage> compilerMessage = new ArrayList<>();
+    private void init(CompilerConfiguration config) {
+        //This parameter is passed in the compiler plugin whether to compile the flow with its dependencies
+        compileWithDependencies = !config.getCustomCompilerArgumentsAsMap().containsKey("ignore-dependencies");
+        //This parameter is used to control the error level. if not set only warnings will be shown
+        errorLevel = config.getCustomCompilerArgumentsAsMap().containsKey("ignore-errors") ? CompilerMessage.Kind.WARNING : CompilerMessage.Kind.ERROR;
```

Comment:
consider breaking long lines (that does not fit review window - it's a good reference point)